src/app/components/Loop/QuoteModal.tsx

```diff
@@ -0,0 +1,264 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Modal, message } from 'antd';
+import { AppState } from 'store/reducers';
+import { getLoopOutQuote, getLoopOut, getLoopIn } from 'modules/loop/actions';
+import { Button, Icon, Alert } from 'antd';
+import { ButtonProps } from 'antd/lib/button';
+import { GetLoopOutArguments, GetLoopInArguments } from 'lib/loop-http/types';
+
+interface StateProps {
+  loopQuote: AppState['loop']['loopQuote'];
+  loop: AppState['loop']['loop'];
+  error: AppState['loop']['error'];
+  hasPassword: boolean;
+}
+
+interface DispatchProps {
+  getLoopOutQuote: typeof getLoopOutQuote;
+  getLoopOut: typeof getLoopOut;
+  getLoopIn: typeof getLoopIn;
+}
+
+interface OwnProps {
+  amt: string;
+  sct: string;
+  isOpen?: boolean;
+  type: string;
+  dest: string;
+  srf: string;
+  sf: string;
+  mf: string;
+  pre: string;
+  chan: string;
+  adv: boolean;
+  htlc: boolean;
+  onClose(): void;
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+class QuoteModal extends React.Component<Props> {
+  componentWillUpdate(nextProps: Props) {
+    if (!this.props.isOpen && nextProps.isOpen) {
+      // Fire even if amt is in store in case we need to cycle
+      this.props.getLoopOutQuote(this.props.amt, this.props.sct);
+    }
+  }
+  render() {
+    const {
+      loopQuote,
+      amt,
+      sct,
+      adv,
+      isOpen,
+      onClose,
+      type,
+      hasPassword,
+      error,
+    } = this.props;
+    if (loopQuote === null) {
+      return null;
+    }
+    const actions: ButtonProps[] = [
+      {
+        children: (
+          <>
+            <Icon type="lightning" theme="filled" /> {`${type}`}
+          </>
+        ),
+        type: 'primary' as any,
+      },
+    ];
+    if (loopQuote === null) {
+      return null;
+    }
+    const isVisible = !!isOpen && !!(hasPassword || error);
+
+    let content;
+    if (loopQuote.miner_fee !== '') {
+      content = (
+        <>
```

Comment:
Unnecessary fragment, it's all in the `QuoteModal` div.

Reply:
removed fragment